pkg/apis/serving/v1alpha1/revision_types.go

```diff
@@ -145,6 +145,15 @@ type RevisionSpec struct {
 	// https://github.com/knative/serving/issues/627
 	// +optional
 	Container corev1.Container `json:"container,omitempty"`
+
+	// TimeoutSeconds defines the request timeout seconds of the revision.
```

Comment:
Can you add a conformance test for this?

We want an app that (like autoscale-go) takes a variable via querystring e.g. `?sleep=X` and takes that long to respond.

We should then deploy two revisions with timeouts `X` and `Y` (far enough apart to measure) and configured with environment variables containing the message to respond with (e.g. `"blue"` and `"green"`).

We should then verify that:
1. Sending messages with `?sleep=N` (`N < min(X, Y)`) get `http.StatusOK`.
1. Sending messages with `?sleep=M` (`M > max(X, Y)`) gets the appropriate HTTP timeout status.
1. Sending messages with `?sleep=O` (`X < O < Y`) gets either a timeout or the message for `Y`.

I'd probably use: `X=2, Y=5, N=0, M=7, O=3`.

Reply:
I created an issue to track the conformance test: https://github.com/knative/serving/issues/1818
I will work on the conformance test in a separate PR.